### PR TITLE
Fix discovery busy-spin

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: pkg/discovery/service.go [hunk 1]
```diff
@@ -129,6 +127,10 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 	}
 
 	for _, node := range list {
+		if ctx.Err() != nil {
```
Comment (mniewrzal) [inline pkg/discovery/service.go]: Just an idea, maybe `Ping` and `UpdateUptime` should check if context was canceled internally?

Comment (egonelbre) [reply pkg/discovery/service.go]: I mean, it should, but I have no idea why doesn't.

Comment (egonelbre) [reply pkg/discovery/service.go]: Ah, found it finally.

Comment (egonelbre) [reply pkg/discovery/service.go]: However, it's still nice to fail early, since we might have a bunch of things in the queue.

Comment (mniewrzal) [reply pkg/discovery/service.go]: ah yes you are right, loop is continuing is case of error.

Comment (egonelbre) [reply pkg/discovery/service.go]: https://github.com/storj/storj/pull/1249 should fix the ping not being canceled issue

**Pull request merged**